internal/config/load.go

```diff
@@ -124,42 +124,75 @@ func mergeAll(fs afero.Fs, repo *git.Repository) (*viper.Viper, error) {
 	return extends, nil
 }
 
-// mergeRemote merges remote config to the current one.
-func mergeRemote(fs afero.Fs, repo *git.Repository, v *viper.Viper) error {
-	var remote Remote
-	err := v.UnmarshalKey("remote", &remote)
+// mergeRemotes merges remotes config to the current one.
+func mergeRemotes(fs afero.Fs, repo *git.Repository, v *viper.Viper) error {
+	var remotes []*Remote
+	var remote *Remote // Use for backward compatibility
+
+	err := v.UnmarshalKey("remotes", &remotes)
 	if err != nil {
 		return err
 	}
 
-	if !remote.Configured() {
-		return nil
+	// Use for backward compatibility
+	err = v.UnmarshalKey("remote", &remote)
+	if err != nil {
+		return err
 	}
 
-	remotePath := repo.RemoteFolder(remote.GitURL)
-	configFile := DefaultConfigName
-	if len(remote.Config) > 0 {
-		configFile = remote.Config
+	// Use for backward compatibility
+	// If "remote" key exists, append it to "remotes"
+	if remote != nil {
+		// Not logged because it's breaking tests
+		// log.Warn("DEPRECATED: \"remote\" key is deprecated, use \"remotes\" instead")
+		remotes = append(remotes, remote)
 	}
-	configPath := filepath.Join(remotePath, configFile)
 
-	log.Debugf("Merging remote config: %s", configPath)
+	for _, remote := range remotes {
+		if !remote.Configured() {
+			continue
+		}
 
-	_, err = fs.Stat(configPath)
-	if err != nil {
-		return nil
-	}
+		// Use for backward compatibility with "remote(s).config" instead of "remote(s).configs"
+		if remote.Config != "" {
+			// Not logged because it's breaking tests
+			// log.Warn("DEPRECATED: \"config\" key is deprecated, use \"configs\" instead for remotes")
+			remote.Configs = append(remote.Configs, remote.Config)
+		}
 
-	if err := merge("remote", configPath, v); err != nil {
-		return err
-	}
+		if len(remote.Configs) == 0 {
+			remote.Configs = append(remote.Configs, DefaultConfigName)
+		}
 
-	if err := extend(v, filepath.Dir(configPath)); err != nil {
-		return err
+		for _, config := range remote.Configs {
+			remotePath := repo.RemoteFolder(remote.GitURL, remote.Ref)
+			configFile := config
+			configPath := filepath.Join(remotePath, configFile)
+
+			log.Debugf("Merging remote config: %s", configPath)
+
+			_, err = fs.Stat(configPath)
+			if err != nil {
+				continue
+			}
+
+			if err = merge("remotes", configPath, v); err != nil {
+				return err
+			}
+
+			if err = extend(v, filepath.Dir(configPath)); err != nil {
+				return err
+			}
```

Comment:
Ok, I've tested this concern and I coundn't reproduce an error, because `extends` get replaced after merging each remote. That's fine. I will polish the PR a little bit and prepare it for the release.